api/src/services/fields.ts

```diff
@@ -65,6 +67,36 @@ export class FieldsService {
 		});
 	}
 
+	async columnInfo(collection?: string): Promise<Column[]>;
+	async columnInfo(collection: string, field: string): Promise<Column>;
+	async columnInfo(collection?: string, field?: string): Promise<Column | Column[]> {
+		const schemaCacheIsEnabled = Boolean(env['CACHE_SCHEMA']);
+
+		let columnInfo: Column[] | null = null;
+
+		if (schemaCacheIsEnabled) {
+			columnInfo = await getCacheValue(this.systemCache, 'columnInfo');
+		}
+
+		if (!columnInfo) {
+			columnInfo = await this.schemaInspector.columnInfo();
+
+			if (schemaCacheIsEnabled) {
+				setCacheValue(this.systemCache, 'columnInfo', columnInfo);
```

Comment:
Could we cache in the `this.localSchemaCache` instead of `systemCache`? The schema cache was updated to be local as the data could be big. 🤔

Reply:
In addition, the `systemCache` is purged when permissions are changed, which is irrelevant for schema caching.

Reply:
switched to the `localSchemaCache` ✅